app/com/m3/octoparts/http/HttpClientPool.scala

```diff
@@ -8,14 +9,28 @@ import scala.util.Try
  * A pool to manage HTTP clients.
  * Holds one HTTP client per partId.
  */
-class HttpClientPool extends KeyedResourcePool[String, HttpClientLike] {
+class HttpClientPool extends KeyedResourcePool[String, HttpPartConfig, HttpClientLike] {
```

Comment:
It seems like this could be a lot simpler if we kept `KeyedResourcePool` as something parameterised on 2 types (key and value), then make a new class called `HttpPartConfigClientKey`, which we can use in the old way (invaliding a keyed resource pool by using keys)

```scala
object HttpPartConfigClientKey {
  def apply(partConfig: HttpPartConfig): HttpPartConfigClientKey // build the key
}

case class HttpPartConfigClientKey(partId: String, httpConnectionTimeout: Duration //other client-related values)
```

Then, in `SimpleHttpHandlerFactory`

```scala
httpClientPool.getOrCreate(HttpPartConfigClientKey(config))
```

and in `CachingRepository`

```scala
httpClientPool.cleanObsolete(cSeq.map(HttpPartConfigClientKey(_))
```

Asides from simpler code, the `KeyedResourcePool` will be smaller in memory-size and need to traverse fewer objects when invalidating things.